### PR TITLE
fix: appointment bugs

### PR DESCRIPTION
**Pull request by LomyW**
- [x] fix scroll https://github.com/corona-school/project-user/issues/771
- [x] decline modal https://github.com/corona-school/project-user/issues/797
- [x] hide create appointment button on dissolved matches https://github.com/corona-school/project-user/issues/798
- [x] image on next appointment on dashboard https://github.com/corona-school/project-user/issues/782
- [x] make appointment card clickable https://github.com/corona-school/project-user/issues/799
- [x] readonly appointments for prospects https://github.com/corona-school/project-user/issues/800
- [x] remove old videochat button https://github.com/corona-school/project-user/issues/801
- [x] join meeting from course detail page https://github.com/corona-school/project-user/issues/804
- [x] increase create appointment modal size https://github.com/corona-school/project-user/issues/805

### CODE REVIEW

File: src/pages/pupil/SingleCoursePupil.tsx [hunk 1]
```diff
@@ -202,7 +202,7 @@ const SingleCoursePupil = () => {
             title: t('single.tabs.lessons'),
             content: (
                 <Box minH={300}>
-                    <AppointmentList isReadOnlyList appointments={data?.subcourse?.appointments as Appointment[]} />
+                    <AppointmentList isReadOnlyList={subcourse?.isParticipant ? false : true} appointments={data?.subcourse?.appointments as Appointment[]} />
```
Comment (timidak) [inline src/pages/pupil/SingleCoursePupil.tsx]:
```suggestion
                    <AppointmentList isReadOnlyList={!subcourse?.isParticipant} appointments={data?.subcourse?.appointments as Appointment[]} />
```

File: src/pages/student/SingleCourseStudent.tsx [hunk 2]
```diff
@@ -231,7 +231,7 @@ const SingleCourseStudent = () => {
             title: t('single.tabs.lessons'),
             content: (
                 <Box minH={300}>
-                    <AppointmentList isReadOnlyList appointments={data?.subcourse?.appointments as Appointment[]} />
+                    <AppointmentList isReadOnlyList={subcourse?.isInstructor ? false : true} appointments={data?.subcourse?.appointments as Appointment[]} />
```
Comment (timidak) [inline src/pages/student/SingleCourseStudent.tsx:234]:
```suggestion
                    <AppointmentList isReadOnlyList={!subcourse?.isInstructor} appointments={data?.subcourse?.appointments as Appointment[]} />
```

File: src/components/MatchAvatarImage.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import PupilAvatar from '../assets/icons/lernfair/avatar_pupil_120.svg';
+import StudentAvatar from '../assets/icons/lernfair/avatar_student_120.svg';
+
+const MatchAvatarImage: React.FC = () => {
+    const userType = useUserType();
+
+    return (
+        <Box>
+            <Center h="100%" w="100%">
+                {userType === 'pupil' ? <StudentAvatar /> : <PupilAvatar />}
```
Comment (timidak) [inline src/components/MatchAvatarImage.tsx:13]: Isn't it the other way around?

Comment (timidak) [reply src/components/MatchAvatarImage.tsx:13]: Ah no, you're desplaying the avatar of the other person, right?

File: src/pages/student/SingleCourseStudent.tsx [hunk 4]
```diff
@@ -354,7 +356,11 @@ const SingleCourseStudent = () => {
                         hideTrafficStatus={canPromoteCourse}
                     />
                     {isInstructorOfSubcourse && !subcourse?.cancelled && !subLoading && (
-                        <StudentCourseButtons subcourse={{ ...subcourse!, ...instructorSubcourse!.subcourse! }} refresh={refetchBasics} />
+                        <StudentCourseButtons
+                            subcourse={{ ...subcourse!, ...instructorSubcourse!.subcourse! }}
```
Comment (timidak) [inline src/pages/student/SingleCourseStudent.tsx:360]: The subcourse and instructor subcourse might not be present (e.g. if an error happens).

File: src/components/MatchAvatarImage.tsx (hunk 3 above)
Comment (LomyW) [reply src/components/MatchAvatarImage.tsx:13]: yes

File: src/components/appointment/AppointmentDetail.tsx [hunk 5]
```diff
@@ -139,6 +139,7 @@ const AppointmentDetail: React.FC<AppointmentDetailProps> = ({ appointment, matc
                     appointmentId={appointment.id}
                     chatType={appointment.appointmentType}
                     isOrganizer={appointment.isOrganizer}
+                    subcoursePublished={appointment.subcourse?.published}
```
Comment (LomyW) [inline src/components/appointment/AppointmentDetail.tsx:142]: check if subcourse of appointment is published

File: src/components/appointment/MetaDetails.tsx [hunk 6]
```diff
@@ -93,10 +96,11 @@ const MetaDetails: React.FC<MetaProps> = ({
                 onPress={() => {
                     navigate(`/video-chat/${appointmentId}/${chatType}`);
                 }}
-                isDisabled={!appointmentId || !canJoinMeeting(startDateTime, duration, isOrganizer ? 30 : 10, DateTime.now())}
+                isDisabled={!subcoursePublished || !appointmentId || !canJoinMeeting(startDateTime, duration, isOrganizer ? 30 : 10, DateTime.now())}
```
Comment (LomyW) [inline src/components/appointment/MetaDetails.tsx:99]: check if subcourse of appointment is published

File: src/pages/Appointment.tsx [hunk 7]
```diff
@@ -37,6 +36,9 @@ export const STUDENT_APPOINTMENT = gql(`
             }
             declinedBy
             zoomMeetingId
+            subcourse {
+                published
```
Comment (LomyW) [inline src/pages/Appointment.tsx:40]: check if subcourse of appointment is published

File: src/pages/Appointments.tsx [hunk 8]
```diff
@@ -45,6 +45,9 @@ const getMyAppointments = gql(`
                 }
                 declinedBy
                 zoomMeetingId
+                subcourse {
```
Comment (LomyW) [inline src/pages/Appointments.tsx:48]: check if subcourse of appointment is published

File: src/pages/student/SingleCourseStudent.tsx [hunk 9]
```diff
@@ -233,7 +233,11 @@ const SingleCourseStudent = () => {
             title: t('single.tabs.lessons'),
             content: (
                 <Box minH={300}>
-                    <AppointmentList isReadOnlyList={!subcourse?.isInstructor} appointments={appointments as Appointment[]} noOldAppointments />
+                    <AppointmentList
+                        isReadOnlyList={!subcourse?.isInstructor || !subcourse.published}
```
Comment (LomyW) [inline src/pages/student/SingleCourseStudent.tsx:237]: check if subcourse of appointment is published

File: src/types/lernfair/Appointment.ts [hunk 10]
```diff
@@ -21,6 +21,9 @@ export type Appointment = {
     isOrganizer?: boolean;
     isParticipant?: boolean;
     zoomMeetingId?: string;
+    subcourse?: {
```
Comment (LomyW) [inline src/types/lernfair/Appointment.ts:24]: add subcourse to appointment type

File: src/widgets/appointment/AppointmentDay.tsx [hunk 11]
```diff
@@ -49,6 +50,7 @@ const AppointmentDay: React.FC<Props> = ({
     displayName,
     appointmentId,
     chatType,
+    subcoursePublished,
```
Comment (LomyW) [inline src/widgets/appointment/AppointmentDay.tsx:53]: check if subcourse of appointment is published

File: src/widgets/appointment/AppointmentDay.tsx [hunk 12]
```diff
@@ -104,6 +106,7 @@ const AppointmentDay: React.FC<Props> = ({
                                 displayName={displayName}
                                 appointmentId={appointmentId}
                                 chatType={chatType}
+                                subcoursePublished={subcoursePublished}
```
Comment (LomyW) [inline src/widgets/appointment/AppointmentDay.tsx:109]: check if subcourse of appointment is published

File: src/widgets/appointment/AppointmentList.tsx [hunk 13]
```diff
@@ -155,6 +155,7 @@ const AppointmentList: React.FC<Props> = ({
                         displayName={appointment.displayName}
                         appointmentId={appointment.id}
                         chatType={appointment.appointmentType}
+                        subcoursePublished={appointment.subcourse?.published}
```
Comment (LomyW) [inline src/widgets/appointment/AppointmentList.tsx:158]: dont show join meeting button, if subcourse not published

File: src/widgets/appointment/AppointmentTile.tsx [hunk 14]
```diff
@@ -99,7 +101,7 @@ const AppointmentTile: React.FC<Props> = ({
                                 </Text>
                             )}
                         </Box>
-                        {!isReadOnly && isCurrentlyTakingPlace && (
+                        {subcoursePublished && !isReadOnly && isCurrentlyTakingPlace && (
```
Comment (LomyW) [inline src/widgets/appointment/AppointmentTile.tsx:104]: dont show join meeting button, if subcourse not published

File: src/components/appointment/MetaDetails.tsx [hunk 15]
```diff
@@ -93,10 +96,11 @@ const MetaDetails: React.FC<MetaProps> = ({
...
             >
                 {t('appointment.detail.videochatButton')}
             </Button>
+            {!subcoursePublished && <AlertMessage content={'Kurs noch nicht veröffentlicht!'} />}
```
Comment (LomyW) [inline src/components/appointment/MetaDetails.tsx:103]: show message if subcourse of appointment is not published

File: src/pages/CreateAppointment.tsx [hunk 16]
```diff
@@ -92,12 +92,14 @@ const CreateAppointment = () => {
                             />
                         )}
                         {currentIndex === 2 && (
-                            <AppointmentCreation
-                                back={noAppointments ? returnToStepOne : onBack}
-                                courseOrMatchId={courseOrMatchId}
-                                isCourse={isCourse}
-                                appointmentsTotal={appointmentsTotal}
-                            />
+                            <Box mt="10">
```
Comment (LomyW) [inline src/pages/CreateAppointment.tsx:95]: add margin

File: src/pages/create-appointment/AppointmentForm.tsx [hunk 17]
```diff
@@ -75,7 +75,7 @@ const AppointmentForm: React.FC<FormProps> = ({ errors, appointmentsCount, onSet
 
     return (
         <Box>
-            <VStack space={5} width="full">
+            <VStack space={1} width="full">
```
Comment (LomyW) [inline src/pages/create-appointment/AppointmentForm.tsx:78]: adjust space

File: src/pages/pupil/Dashboard.tsx [hunk 18]
```diff
@@ -222,14 +222,6 @@ const Dashboard: React.FC<Props> = () => {
         }
     );
 
-    const [joinMeeting, _joinMeeting] = useMutation(
```
Comment (LomyW) [inline src/pages/pupil/Dashboard.tsx:220]: remove old meeting (BBB)

File: src/pages/pupil/SingleCoursePupil.tsx [hunk 19]
```diff
@@ -202,7 +205,11 @@ const SingleCoursePupil = () => {
             title: t('single.tabs.lessons'),
             content: (
                 <Box minH={300}>
-                    <AppointmentList isReadOnlyList={!subcourse?.isParticipant} appointments={data?.subcourse?.appointments as Appointment[]} />
+                    <AppointmentList
+                        isReadOnlyList={!subcourse?.isParticipant}
+                        appointments={data?.subcourse?.appointments as Appointment[]}
+                        noOldAppointments
```
Comment (LomyW) [inline src/pages/pupil/SingleCoursePupil.tsx:211]: dont show load `load old appointments` on course detail page in appointment list

File: src/pages/pupil/single-course/PupilCourseButtons.tsx [hunk 20]
```diff
@@ -144,7 +148,13 @@ const PupilCourseButtons: React.FC<ActionButtonProps> = ({
                         {t('single.actions.contactInstructor')}
                     </Button>
                 )}
-                {subcourse.isParticipant && <JoinMeeting subcourse={subcourse} refresh={refresh} />}
+                {subcourse.isParticipant && (
+                    <JoinMeeting
+                        appointmentId={appointment.id}
```
Comment (LomyW) [inline src/pages/pupil/single-course/PupilCourseButtons.tsx:153]: pass new props to join zoom meeting

File: src/pages/student/SingleCourseStudent.tsx [hunk 21]
```diff
@@ -231,7 +233,7 @@ const SingleCourseStudent = () => {
             title: t('single.tabs.lessons'),
             content: (
                 <Box minH={300}>
-                    <AppointmentList isReadOnlyList={!subcourse?.isInstructor} appointments={data?.subcourse?.appointments as Appointment[]} />
+                    <AppointmentList isReadOnlyList={!subcourse?.isInstructor} appointments={appointments as Appointment[]} noOldAppointments />
```
Comment (LomyW) [inline src/pages/student/SingleCourseStudent.tsx]: dont show load `load old appointments` on course detail page in appointment list

File: src/pages/student/SingleCourseStudent.tsx [hunk 22]
```diff
@@ -354,7 +356,11 @@ const SingleCourseStudent = () => {
                         hideTrafficStatus={canPromoteCourse}
                     />
                     {isInstructorOfSubcourse && !subcourse?.cancelled && !subLoading && (
-                        <StudentCourseButtons subcourse={{ ...subcourse!, ...instructorSubcourse!.subcourse! }} refresh={refetchBasics} />
+                        <StudentCourseButtons
+                            subcourse={{ ...subcourse!, ...instructorSubcourse!.subcourse! }}
+                            refresh={refetchBasics}
+                            appointment={myNextAppointment as Lecture}
```
Comment (LomyW) [inline src/pages/student/SingleCourseStudent.tsx:366]: pass next appointment to join appointment meeting

File: src/pages/student/single-course/StudentCourseButtons.tsx [hunk 23]
```diff
@@ -19,19 +22,26 @@ type SubcourseOfStudent = {
...
+    const canJoin = useMemo(() => {
+        return canJoinMeeting(appointment.start, appointment.duration, 30, DateTime.now());
+    }, [appointment.duration, appointment.start]);
+
     return (
         <>
             <Stack direction={isMobile ? 'column' : 'row'} space={isMobile ? space['1'] : space['2']}>
-                {subcourse.published && <JoinMeeting subcourse={subcourse} isInstructor refresh={refresh} />}
+                {subcourse.published && (
+                    <JoinMeeting isInstructor appointmentId={appointment.id} appointmentType={appointment.appointmentType} canJoinMeeting={canJoin} />
```
Comment (LomyW) [inline src/pages/student/single-course/StudentCourseButtons.tsx:43]: same as pupils

File: src/pages/subcourse/JoinMeeting.tsx [hunk 24]
```diff
@@ -1,79 +1,28 @@
...
-            setDisableMeetingButton(!currentOrNextLecture);
-        }, 1000);
-    }, [subcourse]);
+    const navigate = useNavigate();
 
     return (
         <>
             <Tooltip maxWidth={300} label={isInstructor ? t('course.meeting.hint.student') : t('course.meeting.hint.pupil')}>
-                <Button onPress={getMeetingLink} isDisabled={_joinMeeting.loading || disableMeetingButton}>
+                <Button onPress={() => navigate(`/video-chat/${appointmentId}/${appointmentType}`)} isDisabled={!canJoinMeeting}>
```
Comment (LomyW) [inline src/pages/subcourse/JoinMeeting.tsx:21]: navigate to new zoom meeting page

File: src/components/appointment/MetaDetails.tsx (hunk 15 above)
Comment (timidak) [inline src/components/appointment/MetaDetails.tsx:103]: Maybe adding localization?

File: src/widgets/appointment/AppointmentDay.tsx (hunk 11 above)
Comment (timidak) [inline src/widgets/appointment/AppointmentDay.tsx:53]: isSubcoursePublished?

**Review by timidak — APPROVED** [review]
Looks good so far, just minor adjustments!

File: src/pages/create-appointment/AppointmentCreation.tsx [hunk 25]
```diff
@@ -78,7 +78,7 @@ const AppointmentCreation: React.FC<Props> = ({ back, courseOrMatchId, isCourse,
         } else {
             delete errors.date;
         }
-        if (isCourse && !isDateMinOneWeekLater(appointmentToCreate.date)) {
+        if ((isCourse || isCourseCreation) && !isDateMinOneWeekLater(appointmentToCreate.date)) {
```
Comment (LomyW) [inline src/pages/create-appointment/AppointmentCreation.tsx:81]: problem was, that on the `CourseCreation` the validation didn't pass this point. So an appointment with a date early than one week later could be created -> failed in the backend validation

**Review by timidak — APPROVED** [review]
Looks pretty nice! 🎉

**Pull request merged**